x/ecocredit/core/msg_mint_batch_credits_test.go

```diff
@@ -0,0 +1,40 @@
+package core
+
+import (
+	fmt "fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/thanhpk/randstr"
+)
+
+func TestMsgMintBatchCredits(t *testing.T) {
+	t.Parallel()
+	require := require.New(t)
+	issuer := genAddress()
+
+	tcs := []struct {
+		name string
+		m    MsgMintBatchCredits
+		err  string
+	}{
+		{"invalid issuer", MsgMintBatchCredits{Issuer: "invalid"}, "issuer"},
+		{"invalid denom", MsgMintBatchCredits{Issuer: issuer, BatchDenom: "XXX"}, "invalid denom"},
+		{"invalid note",
+			MsgMintBatchCredits{Issuer: issuer, BatchDenom: batchDenom, Note: randstr.String(514)}, "note must"},
+		{"missing origin tx",
+			MsgMintBatchCredits{Issuer: issuer, BatchDenom: batchDenom}, "origin_tx is required"},
```

Comment:
We should add test cases for invalid origin tx `typ` and `id`.

Reply:
It's in TestValidateOriginTx

Reply:
Thanks for adding https://github.com/regen-network/regen-ledger/pull/991/commits/bf60137c8c1d3a8cfd266a0b3a7abaa270a093ee